src/axom/inlet/Reader.hpp

```diff
@@ -244,6 +245,17 @@ class Reader
   virtual FunctionVariant getFunction(const std::string& id,
                                       const FunctionTag ret_type,
                                       const std::vector<FunctionTag>& arg_types) = 0;
+
+  /*!
+   *****************************************************************************
+   * \brief Retrieves all paths present in the input file
+   *
+   * \return The set of all paths/full names in the input file - this represents
+   * a full traversal of the input file "tree" and includes paths to array/dictionary
+   * entries
+   *****************************************************************************
+   */
+  virtual std::unordered_set<std::string> getAllNames() = 0;
```

Comment:
This seems like an odd approach. I would have expected, for example, each Table to have a list of all the names defined at that level.

Reply:
Should we support both?  Having the Container (new name for Table) hold a list of all names which would allow power users to traverse the tree if they like.  But also have helper functions at a global level?